File: LeetCode/long_pressed_name.cpp
class Solution {
public:
    bool isLongPressedName(string name, string typed) {
        int i, j;
        
        i = j = 0;
        while(i < name.size() && j < typed.size()) {
            if(name[i] == typed[j])
                i++;
            j++;
        }
        
        return i == name.size();
    }
};

File: LeetCode/long_pressed_name.go
func isLongPressedName(name string, typed string) bool {
    i := 0
    var prev byte
    
    for j := 0; j < len(typed); j++ {
        if i < len(name) && name[i] == typed[j] {
            i++
            prev = typed[j]
        } else if typed[j] != prev {
            return false 
        }
    }
    
    return i == len(name)
}
